gcd/maestro_gcd.go

```diff
@@ -102,6 +102,10 @@ func Gcd_init(config *maestroConfig.GcdConfig) {
 		if(err == nil) {
 			b64_config, err := read_config_file(gatewayResource.ConfigFilePath)
 			if  err == nil {
+				// To avoid protocol error, shrink the config resource value to a smaller size
```

Comment:
I'm not sure I understand this, what is it doing?  It seems like it's truncating the Config, which seems like the wrong thing to do.  Should this return an error instead?

Reply:
@costanic The idea is to update the resource value with the content of the config file at the time of initialization. What we found is that when the file size is more than 3.8KB, the object sent to the edge-core over websocket is truncated by the gorilla/websocket library. We do not know why that is or how to fix it so as a hack we are truncating the file size and will report this as known issue.

Reply:
ok sure, but why truncate it again?  Now it must be fixed in two places.  Would it be better to print a warning here saying that since the file size is over 3.8k the message may be truncated, then add a comment in the code here with a link to a jira issue or something.  That way if/when gorilla/websocket is updated and potentially fixed, then this will just start working properly.

Reply:
@costanic Actually the ws api only sends 4096 bytes of the JSON object thus edge core receives a malformed object. That's why we have to reduce the file size before calling write_resource function. And if we try to send more than 4096 bytes then the ws hangs and times out.

Reply:
oh I see.